crates/store/re_chunk/src/chunk.rs

```diff
@@ -240,6 +240,78 @@ impl Chunk {
             .sum()
     }
 
+    /// The cumulative number of events in this chunk for each _unique_ timestamp.
+    ///
+    /// I.e. how many _component batches_ ("cells") were logged in total at each timestamp?
+    ///
+    /// Keep in mind that a timestamp can appear multiple times in a [`Chunk`].
+    /// This method will do a sum accumulation to account for these cases (i.e. every timestamp in
+    /// the returned vector is guaranteed to be unique).
+    #[inline]
+    pub fn num_events_cumulative_per_unique_time(
+        &self,
+        timeline: &Timeline,
+    ) -> Vec<(TimeInt, u64)> {
+        re_tracing::profile_function!();
+
+        let iter_times = || {
+            if self.is_static() {
+                arrow2::Either::Left(std::iter::repeat(TimeInt::STATIC))
+            } else {
+                let times = self.timelines.get(timeline).map_or_else(
+                    || arrow2::Either::Left(std::iter::empty()),
+                    |time_chunk| arrow2::Either::Right(time_chunk.times()),
+                );
+                arrow2::Either::Right(times)
+            }
+        };
+
+        // NOTE: This is used on some very hot paths (time panel rendering).
+        // Performance trumps readability. Optimized empirically.
+
+        // Raw, potentially duplicated counts (because timestamps aren't necessarily unique).
+        let mut counts_raw = vec![0u64; self.num_rows()];
+        {
+            self.components.values().for_each(|list_array| {
+                if let Some(validity) = list_array.validity() {
+                    validity
+                        .iter()
+                        .enumerate()
+                        .for_each(|(i, is_valid)| counts_raw[i] += is_valid as u64);
+                } else {
+                    counts_raw.iter_mut().for_each(|count| *count += 1);
+                }
+            });
+        }
+
+        let mut counts = Vec::with_capacity(counts_raw.len());
+
+        let Some(mut cur_time) = iter_times().next() else {
+            return Vec::new();
+        };
+        let mut cur_count = 0;
+        izip!(iter_times(), counts_raw).for_each(|(time, count)| {
```

Comment:
⚠️ This only works if the times are already sorted, which this method does not ensure

Reply:
yep, nice catch